tests/testTriggers.go

```diff
@@ -813,6 +814,134 @@ func TriggerDeployNewApps(contexts *[]*scheduler.Context, recordChan *chan *Even
+				stcList, err := pxOperator.ListStorageClusters(namespace)
+				if err != nil {
+					log.InfoD("Storage cluster not available")
+					UpdateOutcome(event, err)
+				}
+				log.InfoD("Stc list %v ",stcList)
+				storageNodes:= node.GetStorageNodes()
+				stc, err := pxOperator.GetStorageCluster(stcList.Items[0].Name, stcList.Items[0].Namespace)
+				if err != nil {
+					UpdateOutcome(event, err)
+				}
+				var nodeId string
+				nodeId = storageNodes[0].VolDriverNodeID
+				err = Inst().N.DetachDrivesFromVM(stc, storageNodes[0].Name)
+				if err != nil {
+					//log.InfoD("Failed to detach cloud drives %v", storageNodes[0].Name)
+					UpdateOutcome(event, err)
+				}
+				time.Sleep(time.Duration(1 * time.Minute))
+				statusErr := Inst().V.WaitDriverUpOnNode(storageNodes[0], 10*time.Minute)
+				if statusErr != nil {
+					if strings.Contains(statusErr.Error(), apios.Status_STATUS_STORAGE_DOWN.String()) {
+						log.InfoD("Node has gone to Storage Down state after detach %v: Node %s", statusErr, storageNodes[0].Name)
+					}else{
+						UpdateOutcome(event, statusErr)
+					}
+				}
+				storageNode, err := Inst().V.GetDriverNode(&storageNodes[0])
+				if err != nil{
+					UpdateOutcome(event, err)
+				}
+				if storageNode.Status.String() != apios.Status_STATUS_STORAGE_DOWN.String(){
+					UpdateOutcome(event, fmt.Errorf("Node %s: Expected: %v Actual: %v", storageNode.SchedulerNodeName,
+					apios.Status_STATUS_STORAGE_DOWN, storageNode.Status))
+				}
+				log.InfoD("Status of the storage node %s ,%v ", storageNode.SchedulerNodeName, storageNode.Status)
+				statusErr = Inst().V.EnterMaintenance(storageNodes[0])
+				if statusErr != nil {
+					log.Infof("Status after entermaintenance node %s : %v", storageNodes[0].Name,statusErr)
+					UpdateOutcome(event, statusErr)
+				}
+				status, _ := Inst().V.GetNodeStatus(storageNodes[0])
+				log.InfoD("Status when the storage node entered maintenance mode %s , ", status.String())
+				if status.String() != apios.Status_STATUS_MAINTENANCE.String(){
+					UpdateOutcome(event, fmt.Errorf("Node %s: Expected: %v Actual: %v", storageNodes[0].Name,
+					apios.Status_STATUS_MAINTENANCE, status))
+				}
+				statusErr = Inst().V.ExitMaintenance(storageNodes[0])
+				if statusErr != nil {
+					log.InfoD("Status after exit maintenance mode node %s : %v", storageNodes[0].Name, statusErr)
+				}
+				status, _ = Inst().V.GetNodeStatus(storageNodes[0])
+				log.InfoD("Node Status after exit maintenance mode %v , %s", status, storageNodes[0].Name)
+				if status.String() != apios.Status_STATUS_OK.String(){
+					UpdateOutcome(event, fmt.Errorf("Node %s: Expected: %v Actual: %v", storageNodes[0].Name,
+					apios.Status_STATUS_MAINTENANCE, status))
+				}
+				//There are chances that storageless node can convert storage node by attaching these drives. Hence making sure
+				//This node id is up.
+				var selectedStorageNode *node.Node
+				storageNodes = node.GetStorageNodes()
```

Comment:
After detaching and doing doing node maintenance cycle we need to refresh the driver endpoints.
here node.GetStorageNodes() will have the same data as before tests start as it is not refreshed